release notes script changes

bdhirsh opened the pull request:
Added a few QoL improvements to the release notes categorization script.

I added a few extra filters, that caught and auto-labeled around 800-1000 extra commits.

The process for re-running the script with my filters was a bit painful, so I also added a basic `--re-run-with-filters` flag to the script, that looks at the previous output, and tries to update any commits that were previously "Uncategorized" using the new filters.

Stack from [ghstack](https://github.com/ezyang/ghstack):
* #75020
* __->__ #72154

bdhirsh committed:
release notes script changes

[ghstack-poisoned]

bdhirsh committed:
Update on "release notes script changes"




[ghstack-poisoned]

bdhirsh committed:
Update on "release notes script changes"




[ghstack-poisoned]

bdhirsh commented:
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34088700).

bdhirsh committed:
Update on "release notes script changes"


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

bdhirsh commented:
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34088700).

bdhirsh committed:
Update on "release notes script changes"


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

bdhirsh commented:
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34088700).

bdhirsh committed:
Update on "release notes script changes"


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

bdhirsh commented:
@bdhirsh has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34088700).

bdhirsh committed:
Update on "release notes script changes"


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

bdhirsh committed:
Update on "release notes script changes"


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

bdhirsh commented on scripts/release_notes/commitlist.py:
```diff
@@ -129,7 +129,7 @@ def categorize(commit_hash, title):
                 break
             # torch/quantization, test/quantization, aten/src/ATen/native/quantized, torch/nn/{quantized, quantizable}
-            if CommitList.keywordInFile(file, ['torch/quantization', 'test/quantization', 'aten/src/ATen/native/quantized', 'torch/nn/quantiz']):
+            if CommitList.keywordInFile(file, ['torch/quantization', 'test/quantization', 'aten/src/ATen/native/quantized', 'torch/nn/quantiz', 'torch/ao/quantization']):
```
There's also a `torch/ao/sparsity` (which I added below under the sparse file paths). When I saw that folder I assumed that `torch/ao` includes stuff aside from quantization?

albanD replied:
It is.
We should actually check with them if they want all ao as a single category or not.

bdhirsh replied:
From discussion with @vkuzo, the AO team would like an "AO" tag. I'll update the PR to map `torch/ao` to the AO tag (and in the next release, we'll probably want to lump the PR's tagged with the existing "quantization" tag together with "AO).

bdhirsh committed:
Update on "release notes script changes"

Added a few QoL improvements to the release notes categorization script.

I added a few extra filters, that caught and auto-labeled around 800-1000 extra commits.

The process for re-running the script with my filters was a bit painful, so I also added a basic `--re-run-with-filters` flag to the script, that looks at the previous output, and tries to update any commits that were previously "Uncategorized" using the new filters.


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

albanD reviewed (approved):
SGTM

anjali411 commented on scripts/release_notes/commitlist.py:
```diff
@@ -252,6 +280,11 @@ def main():
     group.add_argument('--create_new', nargs=2)
     group.add_argument('--update_to')
+    # I found this flag useful when experimenting with adding new auto-categorizing filters.
+    # After running commitlist.py the first time, if you add any new filters in this file,
```
what do you mean by filter here?

bdhirsh replied:
Just adding new directories that map directly to specific categories. I was experimenting with adding new ones, so this change made it convenient to add a new directory -> category mapping, and see which PR's got newly labeled because of it.

anjali411 commented on scripts/release_notes/commitlist.py:
```diff
@@ -128,6 +128,9 @@ def categorize(commit_hash, title):
             if CommitList.keywordInFile(file, ['torch/fx', 'test_fx']):
                 category = 'fx'
                 break
+            if CommitList.keywordInFile(file, ['torch/ao', 'test/ao']):
```
since now we `release notes: ` and `topic: ` tags, we should add a small helper to infer the tags automatically before entering this for loop, and only enter this loop if the PR didn't have the `release notes :` tag.

bdhirsh replied:
I think the next PR that I added does that? https://github.com/pytorch/pytorch/pull/75020/files

bdhirsh committed:
Update on "release notes script changes"

Added a few QoL improvements to the release notes categorization script.

I added a few extra filters, that caught and auto-labeled around 800-1000 extra commits.

The process for re-running the script with my filters was a bit painful, so I also added a basic `--re-run-with-filters` flag to the script, that looks at the previous output, and tries to update any commits that were previously "Uncategorized" using the new filters.


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]

bdhirsh committed:
Update on "release notes script changes"

Added a few QoL improvements to the release notes categorization script.

I added a few extra filters, that caught and auto-labeled around 800-1000 extra commits.

The process for re-running the script with my filters was a bit painful, so I also added a basic `--re-run-with-filters` flag to the script, that looks at the previous output, and tries to update any commits that were previously "Uncategorized" using the new filters.


Differential Revision: [D34088700](https://our.internmc.facebook.com/intern/diff/D34088700)

[ghstack-poisoned]
